RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress+View.swift

```diff
@@ -83,23 +101,48 @@ public extension AccountRecoveryScanInProgress {
 				.presentsLoadingViewOverlay()
 				.padding()
 				.footer {
+					// FIXME: Strings
+					Button("Tap here to scan the next \(batchSize)") {
+						store.send(.view(.scanMore))
+					}.buttonStyle(.alternativeRectangular)
+
+					// FIXME: Strings
 					Button("Continue") {
 						store.send(.view(.continueTapped))
 					}.buttonStyle(.primaryRectangular)
 				}
-				.sheet(
-					store: store.scope(
-						state: \.$derivePublicKeys,
-						action: { .child(.derivePublicKeys($0)) }
-					),
-					content: {
-						DerivePublicKeys.View(store: $0)
-					}
-				)
-				.onFirstTask { @MainActor in
-					await store.send(.view(.onFirstTask)).finish()
+				.onFirstAppear {
+					viewStore.send(.onFirstAppear)
 				}
+				.destination(store: store)
 			}
 		}
 	}
 }
+
+private extension StoreOf<AccountRecoveryScanInProgress> {
+	var destination: PresentationStoreOf<AccountRecoveryScanInProgress.Destination> {
+		func scopeState(state: State) -> PresentationState<AccountRecoveryScanInProgress.Destination.State> {
+			state.$destination
+		}
+		return scope(state: scopeState, action: Action.destination)
+	}
+}
+
+@MainActor
+private extension View {
+	func destination(store: StoreOf<AccountRecoveryScanInProgress>) -> some View {
```

Comment:
I call these destinationS, even if there's only one. partly because everything else is called destination in singular.